ast/compile.go

```diff
@@ -2946,6 +2958,10 @@ func reorderBodyForClosures(arity func(Ref) int, globals VarSet, body Body) (Bod
 			vs := VarSet{}
 			WalkClosures(e, func(x interface{}) bool {
 				vis := &VarVisitor{vars: vs}
+				if ev, ok := x.(*Every); ok {
+					vis.Walk(ev.Body)
+					return true
+				}
```

Comment:
This is a bit of a trap: `WalkClosures` checks that its node in question is of type *Every. So, when passing a function, you get an `x` of type *Every, and to walk its actual closure, the `every.Body`, it'll have to be done like this.

I wonder if there's a better approach here. Having `WalkClosures` apply `f` to `ev.Body` [here](https://github.com/open-policy-agent/opa/pull/4231/files#diff-0231d068c880c5fa17a560d427a97e6a8a2e4fc8c6ba048e4427baa7355b67ccR165) would not allow us to differentiate what we've been called on when we need it, [here](https://github.com/open-policy-agent/opa/pull/4231/files#diff-067a842cc5e1a348551f4c7656120105acedafb2faac6e7b7ff57b75c5349570R1466-R1468).

Reply:
I think this is unavoidable due to the scoping rules we've implemented for `every`. Presumably it'll be the same for any other keywords we add that support closures and local variable declaration. This seems like the best we can do.